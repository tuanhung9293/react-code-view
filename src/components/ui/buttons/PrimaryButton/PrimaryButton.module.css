@value colors: 'styles/colors.css';
@value secondary, danger from colors;

.base{
  box-sizing: border-box;
  display: inline-block;
  font-size: 18px;
  font-weight: bold;
  color: white;
  background-color: secondary;
  text-decoration: none;
  padding: 6px 22px;
  text-transform: none;
  border: 2px solid secondary;
  border-radius: 3px;
  text-align: center;
  cursor: pointer;
}

.base:hover:not(.disabled), .base:focus:not(.disabled){
  outline: none;
  background-color: #83A8FF;
  border-color: #83A8FF;
}

.base:active:not(.disabled){
  background-color: #3D5FB0;
  border-color: #3D5FB0;
}

.base.disabled{
  opacity: 0.5;
  background-color: primary;
  cursor: default;
}

.outlined{
  border-color: black;
  color: black;
  background-color: transparent;
}

.outlined:hover:not(.disabled), .outlined:focus:not(.disabled){
  background-color: rgba(0,0,0,0.05);
  border-color: black;
}

.outlined:active:not(.disabled){
  background-color: rgba(0,0,0,0.15);
  border-color: black;
}

.outlined.danger{
  border-color: danger;
  color: danger;
}

.outlined.danger:hover:not(.disabled), .outlined.danger:focus:not(.disabled){
  background-color: #F2DADA;
  border-color: danger;
}

.outlined.danger:active:not(.disabled) {
  background-color: #E9C0C0;
  border-color: danger;
}

.wide{
  width: 100%;
  margin: 10px 0;
}
