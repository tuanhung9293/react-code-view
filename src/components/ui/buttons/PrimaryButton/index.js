import React from 'react'
import classNames from 'classnames'
import Button from '../Button'
import styles from './PrimaryButton.module.css'

function PrimaryButton({disabled, outlined, danger, wide, ...props}) {
  const className = classNames({
    [styles.base]: true,
    [styles.disabled]: disabled,
    [styles.outlined]: outlined,
    [styles.danger]: danger,
    [styles.wide]: wide
  })
  return <Button {...props} disabled={disabled} className={className} />
}

export default PrimaryButton