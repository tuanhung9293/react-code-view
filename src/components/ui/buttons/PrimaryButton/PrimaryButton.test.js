import React from 'react'
import {shallow} from 'enzyme'
import PrimaryButton from './'

it('renders as a button with children', () => {
  expect(shallow(
    <PrimaryButton>I am child</PrimaryButton>
  )).toMatchSnapshot()
})

it('overrides className', () => {
  expect(shallow(
    <PrimaryButton className="i-should-not-be-there">I am child</PrimaryButton>
  )).toMatchSnapshot()
})

it('renders as disabled', () => {
  expect(shallow(
    <PrimaryButton disabled>I am child</PrimaryButton>
  )).toMatchSnapshot()
})

it('renders as outlined', () => {
  expect(shallow(
    <PrimaryButton outlined>I am child</PrimaryButton>
  )).toMatchSnapshot()
})

it('renders as outlined danger', () => {
  expect(shallow(
    <PrimaryButton outlined danger>I am child</PrimaryButton>
  )).toMatchSnapshot()
})