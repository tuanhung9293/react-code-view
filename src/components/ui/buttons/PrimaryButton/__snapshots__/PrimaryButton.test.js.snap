// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overrides className 1`] = `
<Button
  className="base"
>
  I am child
</Button>
`;

exports[`renders as a button with children 1`] = `
<Button
  className="base"
>
  I am child
</Button>
`;

exports[`renders as disabled 1`] = `
<Button
  className="base disabled"
  disabled={true}
>
  I am child
</Button>
`;

exports[`renders as outlined 1`] = `
<Button
  className="base outlined"
>
  I am child
</Button>
`;

exports[`renders as outlined danger 1`] = `
<Button
  className="base outlined danger"
>
  I am child
</Button>
`;
